apps/api/src/controllers/applicationController.ts

```diff
@@ -257,6 +257,44 @@ export const submitRevision = async ({ applicationId }: { applicationId: number
 	}
 };
 
+export const revokeApplication = async (applicationId: number): AsyncResult<ApplicationRecord> => {
+	try {
+		// Fetch application
+		const database = getDbInstance();
+		const service: ApplicationService = applicationSvc(database);
+		const result = await service.getApplicationById({ id: applicationId });
+
+		if (!result.success) {
+			return result;
+		}
+
+		const application = result.data;
+
+		const appStateManager = new ApplicationStateManager(application);
+
+		if (appStateManager.state !== ApplicationStates.APPROVED) {
+			return failure('Application should be in APPROVED status', 'RevokeConflict');
+		}
```

Comment:
All the logic for what states are valid to revoke from are stored in the state manage, we don't need this check, we only need the next line's `appStateManager.revokeApproval()`.